src/server/api/routers/company.ts

```diff
@@ -0,0 +1,77 @@
+import { z } from "zod";
+import { createTRPCRouter, publicProcedure } from "~/server/api/trpc";
+import { Industry } from "@prisma/client";
+
+export const companyRouter = createTRPCRouter({
+  getAll: publicProcedure.query(({ ctx }) => {
+    return ctx.db.company.findMany();
+  }),
+  getByName: publicProcedure
+    .input(
+      z.object({
+        name: z.string(),
+      }),
+    )
+    .query(({ ctx, input }) => {
+      return ctx.db.company.findFirst({
+        where: {
+          name: input.name,
+        },
+      });
+    }),
+  deleteByID: publicProcedure
+    .input(
+      z.object({
+        id: z.string(),
+      }),
+    )
+    .mutation(({ ctx, input }) => {
+      // TODO : should we delete all of the roles + reviews associated with a company?
+      return ctx.db.company.delete({
+        where: {
+          id: input.id,
+        },
+      });
+    }),
+  postCompany: publicProcedure
```

Comment:
While `GET` requests to access the companies is public (given that people can view reviews without signing in -- again I don't believe we have formally established this behavior but I _feel_ that this would be the natural approach), `POST`, `PUT`, and `DELETE` operations must only be allowed by authenticated users.

Try using `protectedProcedure` from `~/server/api/trpc` to ensure that only authenticated users can call this endpoint. Anyone who is not authenticated would be presented with a `403 Forbidden` error by `tRPC`, I believe.

Testing this would be hard for _now_ since we don't have OAuth set up. I will work on that in the meantime so that we can use NextAuth to create users, in turn allowing us to call protected endpoints.